### PR TITLE
[Feat.]Executor feature to delete jobs after completion

### PR DESCRIPTION
**Pull request by rahulchheda**
Signed-off-by: Rahul M Chheda <rahul.chheda@mayadata.io>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
https://github.com/litmuschaos/litmus/issues/857

**Checklist**

* [ ] Does this PR have a corresponding GitHub issue?
* [ ] Have you included relevant README for the chaoslib/experiment with details?
* [ ] Have you added debug messages where necessary? 
* [ ] Have you added task comments where necessary? 
* [ ] Have you tested the changes for possible failure conditions?
* [ ] Have you provided the positive & negative test logs for the litmusbook execution?
* [ ] Does the litmusbook ensure idempotency of cluster state?, i.e., is cluster restored to original state?
* [ ] Have you used non-shell/command modules for Kubernetes tasks?
* [ ] Have you (jinja) templatized custom scripts that is run by the litmusbook, if any? 
* [ ] Have you (jinja) templatized Kubernetes deployment manifests used by the litmusbook, if any?
* [ ] Have you reused/created util functions instead of repeating tasks in the litmusbook?
* [ ] Do the artifacts follow the appropriate directory structure? 
* [ ] Have you isolated storage (eg: OpenEBS) specific implementations, checks? 
* [ ] Have you isolated platform (eg: baremetal kubeadm/openshift/aws/gcloud) specific implementations, checks?  
* [ ] Are the ansible facts well defined? Is the scope explicitly set for playbook & included utils?
* [ ] Have you ensured minimum/careful usage of shell utilities (awk, grep, sed, cut, xargs etc.,)?
* [ ] Can the limtusbook be executed both from within & outside a container (configurable paths, no hardcode)?
* [ ] Can you suggest the minimal resource requirements for the litmusbook execution?
* [ ] Does the litmusbook job artifact carry comments/default options/range for the ENV tunables?
* [ ] Has the litmusbooks been linted? 

**Special notes for your reviewer**:

### CODE REVIEW

File: executor/executor.yml [hunk 1]
```diff
@@ -218,4 +218,19 @@
 
 - debug:
     msg: "engine : {{ c_engine_yaml.stdout }}" 
- 
\ No newline at end of file
+
+- name: Register the Job Name for further use
```
[the remark this answers is not in the document]
Comment (ksatchit) [reply executor/executor.yml]: Sorry for the nit-pick again: Can we do this _before_ the job execution (one stop before)

[the remark this answers is not in the document]
Comment (rahulchheda) [reply executor/executor.yml]: No problem sir, I have made the changes please check

File: executor/executor.yml [hunk 2]
```diff
@@ -218,4 +224,16 @@
...
+  shell: >
+    kubectl get chaosengine {{ c_engine }} -n {{ c_app_ns }} -o jsonpath='{.spec.jobCleanUpPolicy}'
+  args:
+    executable: /bin/bash
+  register: jobCleanUpPolicy
+
+- name: Delete job if the jobCleanUpPolicy status is delete
+  shell: >
+    kubectl delete job {{ job_name }} -n {{ c_app_ns }} 
+  when: jobCleanUpPolicy.stdout == "delete"
```
Comment (Sanjay1611) [inline executor/executor.yml]: I think this check will retain the completed jobs even if `jobCleanUpPolicy` is something other than `retain`. Will it be ok @ksatchit @chandankumar4

Comment (ksatchit) [reply executor/executor.yml]: There are a few more such scenarios where we will need stricter validation of the values provided in the schema. Raised issue #878 for this. @rahulchheda could you please enhance the executor README with note regarding the "supported" values, i.e., (example chaosEngine with this option specified)

Comment (Sanjay1611) [reply executor/executor.yml]: Thank you @rahulchheda @ksatchit .

**Pull request merged**